content/client-lib-development-guide/features.textile

```diff
@@ -299,14 +300,19 @@ h3(#realtime-connection). Connection
 ** @(RTN10c)@ The last known @ProtocolMessage@ @connectionSerial@ is used when restoring connection state
 * @(RTN11)@ @Connection#connect@ function:
 ** @(RTN11a)@ Explicitly connects to the Ably service if not already connected
-** @(RTN11b)@ An error will be indicated if the state is @CLOSING@ as the connection must complete the close request before reconnecting
+** @(RTN11b)@ If the state is @CLOSING@, the client should make a new connection with a new transport instance and remove all references to the old one. In particular, it should make sure that, when the @CLOSED@ @ProtocolMessage@ arrives for the old connection, it doesn't affect the new one.
+** @(RTN11c)@ If the state is @DISCONNECTED@ or @SUSPENDED@, aborts the retry process described in "RTN14d":#RTN14d and "RTN14e":#RTN14e and immediately tries to reconnect.
+** @(RTN11d)@ If the state is @FAILED@, moves all the channels to @INITIALIZED@, sets their @errorReason@ to null, and sets the connection's @errorReason@ to null.
 * @(RTN12)@ @Connection#close@ function:
-** @(RTN12a)@ Sends a @CLOSE@ @ProtocolMessage@ to the server, sets the state to @CLOSING@ and waits for a @CLOSED@ @ProtocolMessage@ to be received
+** @(RTN12f)@ If the connection state is @CONNECTING@, do the operation once the connection state is @CONNECTED@
```

Comment:
> So while this does improve behaviour in the case of a slow-but-reliable network (ie where there's a larger chance of the close() happening between the connect request but before the CONNECTED), it's at the expense of making it seem unresponsive when the network is failing/unreliable. :/

Well, yes, but if they don't care about closing connections cleanly then they would never need to call `close()`; they just exit. So if they are calling `close()` then they are making a good faith attempt to release the resources associated with the connection, so why not do that?

At the expense of further complexity we might say that the wait only occurs if it is a `resume` and not a new connection?

I did suggest the possibility of a shorter timeout